src/macos/ATDriverGenericMacOS/ATDriverGenericService/ATDriverClientUnix.swift

```diff
@@ -0,0 +1,87 @@
+//
+//  ATDriverClientUnix.swift
+//  ATDriverGenericMacOS
+//
+//  Created by Z Goddard on 1/4/24.
+//
+
+import Foundation
+
+class ATDriverClientUnix {
+
+  class SocketResource {
```

Comment:
Made this RAII type to automatically handle closing the socket. In the case of how its currently used, when the resource goes out of scope in `_sendEvent`, it'll de-initialize and close the socket. This way we don't have to be careful of all the edge cases around throw, return, and other scenarios of the socket left hanging open.